editor/editor_node.cpp

```diff
@@ -1218,7 +1218,15 @@ Error EditorNode::load_resource(const String &p_resource, bool p_ignore_broken_d
 
 	Ref<Resource> res;
 	if (ResourceLoader::exists(p_resource, "")) {
+		if (p_ignore_broken_deps) {
+			ResourceLoader::set_abort_on_missing_resources(true);
```

Comment:
Are you sure this shouldn't be called with `false`?

`false` would cause `ResourceLoader::notify_dependency_error()` to be called instead of aborting and returning an error code. And by `p_ignore_broken_deps == true`, the branch below to open the deps dialog won't be run.

Reply:
`false` is the default value.
I tested and `true` is the correct one to use. That's why I said it's backwards 🙃

Reply:
Then please help me follow the logic, because I think I'm misunderstanding it at some point.

1. `EditorNode::load_resource()` is called with `p_ignore_broken_deps = true`. So, what you want is the thing to be loaded with missing external resources as `null` and no dependency error dialog.
1. `ResourceLoader::set_abort_on_missing_resources(true);`
1. `ResourceLoaderText` fails to load an external resource and checks `ResourceLoader::get_abort_on_missing_resources()`.
1. Since it's `true`, it returns `ERR_FILE_MISSING_DEPENDENCIES` and cancels further loading. That's not what we wanted, is it?

Alternatively:
1. Same as before.
1. `ResourceLoader::set_abort_on_missing_resources(false);`
1. `ResourceLoaderText` fails to load an external resource and checks `ResourceLoader::get_abort_on_missing_resources()`.
1. Since it's `false`, it reports the broken dep via `notify_dependency_error()`. `EditorNode::load_resource()` will ignore `dependency_errors` because `p_ignore_broken_deps == true` and won't show the broken deps dialog.

What am I missing here?

Reply:
https://github.com/godotengine/godot/blob/7022271291a3d2a9cbd6a223d22a29fd775dfc5d/scene/resources/resource_format_text.cpp#L278-L290
When the parser returns any other error than ERR_FILE_EOF or ERR_FILE_MISSING_DEPENDENCIES (the latter appears with abort = true), the loader will return empty scene.

Not sure about binary format, the issue still persists for binary scenes it seems.